src/main/java/space/obminyashka/items_exchange/service/impl/AdvertisementServiceImpl.java

```diff
@@ -85,6 +85,15 @@ public Page<AdvertisementTitleDto> findByKeyword(String keyword, Pageable pageab
         return Page.empty();
     }
 
+    @Override
+    public Page<AdvertisementTitleDto> findByCategoryId(Long categoryId, Pageable pageable) {
+        final var searchResult = advertisementRepository.findAdvertisementByCategoryId(categoryId, pageable);
+        if (!searchResult.isEmpty()) {
+            return searchResult.map(this::buildAdvertisementTitle);
+        }
+        return Page.empty();
```

Comment:
That code will do the same
```suggestion
        return advertisementRepository.findAdvertisementByCategoryId(categoryId, pageable)
                   .map(this::buildAdvertisementTitle);
```

Reply:
Done